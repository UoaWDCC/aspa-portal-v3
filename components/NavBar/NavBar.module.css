.header {
    background-color: #717882;
}

.root {
    position: relative;
    width: fit-content;
    border-radius: var(--mantine-radius-md);
    height: 5vw;
}

.control {
    line-height: 1;
    color: var(--mantine-color-white);
    border-radius: var(--mantine-radius-md);
    font-size: var(--mantine-font-size-sm);
    transition: color 100ms ease;
    font-weight: 500;
    font-size: 1.5vw;
    width: 8vw;
    height: 5vw;
    text-align: center;
    padding-top: 2%;

    @mixin hover {
        font-weight: 1000;
    }

    &[data-active] {
        color: var(--mantine-color-white);
    }
}

.controlLabel {
    position: relative;
    z-index: 1;
}

.logoCircle {
  width: 2.5vw;
  height: 2.5vw;
  border-radius: 50%;
  object-fit: cover;
  cursor: pointer;
}

@keyframes roll {
  0% {
    transform: translateX(0) rotate(0deg);
  }
  100% {
    transform: translateX(var(--mantine-indicator-x, 0px)) rotate(360deg);
  }
}

.indicator {
  background: black;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
  font-weight: bold;
  font-size: 0.75rem;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.3);
  position: relative;
  bottom: 15%;
  left: 40%;
  z-index: 10;
}

.indicatorRotatingWrapper {
  transform: rotate(var(--rotation));
  transition: transform 0.4s ease-in-out;
}
