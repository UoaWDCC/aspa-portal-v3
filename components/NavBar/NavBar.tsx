import { useState } from "react";
import {
  AppShell,
  Group,
  Burger,
  UnstyledButton,
  NavLink,
  FloatingIndicator,
} from "@mantine/core";
import { useDisclosure } from "@mantine/hooks";
import Link from "next/link";
import styles from "./NavBar.module.css";

interface Link {
  name: string;
  href: string;
}

interface NavBarProps {
  links: Link[];
}

export function NavBar({ links }: NavBarProps) {
  const [opened, { toggle }] = useDisclosure();
  const [active, setActive] = useState(0);

  const [rootRef, setRootRef] = useState<HTMLDivElement | null>(null);
  const [controlsRefs, setControlsRefs] = useState<
    Record<string, HTMLButtonElement | null>
  >({});

  const setControlRef = (index: number) => (node: HTMLButtonElement) => {
    controlsRefs[index] = node;
    setControlsRefs(controlsRefs);
  };

  const controls = links.map((link, index) => (
    <Link key={index} href={link.href} passHref legacyBehavior>
      <UnstyledButton
        key={index}
        className={styles.control}
        onClick={() => setActive(index)}
        ref={setControlRef(index)}
        mod={{ active: active === index }}
      >
        <div
          className={`${styles.controlName} ${index === active ? styles.hidden : ""}`}
        >
          {link.name}
        </div>
      </UnstyledButton>
    </Link>
  ));

  return (
    <AppShell
      navbar={{
        width: 300,
        breakpoint: "sm",
        collapsed: { desktop: true, mobile: !opened },
      }}
      padding="md"
    >
      <AppShell.Header className={styles.header}>
        <Group h="100%" px="md">
          <Burger opened={opened} onClick={toggle} hiddenFrom="sm" size="sm" />
          <Group
            justify="space-between"
            style={{ flex: 1, justifyContent: "flex-end" }}
          >
            <Group ml="xl" gap={20} visibleFrom="sm">
              <div className={styles.root} ref={setRootRef}>
                {controls}
                <FloatingIndicator
                  target={controlsRefs[active]}
                  parent={rootRef}
                  className={styles.indicator}
                />
              </div>
            </Group>
          </Group>
        </Group>
      </AppShell.Header>

      <AppShell.Navbar py="md" px={4}>
        {links.map((link, index) => (
          <NavLink
            key={index}
            label={link.name}
            href={link.href}
            active={index === active}
            onClick={() => setActive(index)}
          />
        ))}
      </AppShell.Navbar>
    </AppShell>
  );
}
