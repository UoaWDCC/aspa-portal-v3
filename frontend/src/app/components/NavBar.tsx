import styles from './Navbar.module.css'

export default function NavBar() {
    return (
        <>
            <div className={styles.register}>

                <ul className={styles.nav}>

                    <li className={styles.navitem}>
                        <a className={styles.navlink} href="/"> Home </a>
                    </li>

                    <li className={styles.navitem}>
                        <a className={styles.navlink} href="/"> Talk To Us </a>
                    </li>

                    <li className={styles.navitem}>
                        <a className={styles.navlink} href="/"> Event </a>
                    </li>

                    <li className={styles.navitem}>
                        <a className={styles.navlink} href="/"> Profile </a>
                    </li>

                    <li className={styles.navitem}>
                        <a className={styles.navlink} href="/"> Member Data </a>
                    </li>

                </ul>

            </div>
        </>
    )
}