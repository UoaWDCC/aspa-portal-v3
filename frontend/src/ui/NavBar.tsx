'use client';

import Link from 'next/link'
import styles from './Navbar.module.css'
import { usePathname } from 'next/navigation';

export default function NavBar() {
    const pathname = usePathname();
    console.log(pathname)

    return (
        <nav className={styles.nav}>
            <ul>
                <li className={styles.navitem}>
                    <Link className={`${styles.navlink} ${pathname === "/home" ? styles.currentpage : ""}`} href="/home">
                        Home
                    </Link>
                </li>
                <li className={styles.navitem}>
                    <Link className={`${styles.navlink} ${pathname === "/contact" ? styles.currentpage : ""}`} href="/contact">
                        Talk To Us
                    </Link>
                </li>
                <li className={styles.navitem}>
                    <Link className={`${styles.navlink} ${pathname === "/events" ? styles.currentpage : ""}`} href="/events">
                        Events
                    </Link>
                </li>
                <li className={styles.navitem}>
                    <Link className={`${styles.navlink} ${pathname === "/home/login" ? styles.currentpage : ""}`} href="/home/login">
                        Register/Log In
                    </Link>
                </li>
            </ul>
        </nav>
    )
}
