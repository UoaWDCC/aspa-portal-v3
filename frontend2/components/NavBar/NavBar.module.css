.header {
    background-color: var(--mantine-color-aspa-orange-4);
    height: 60px;
}

.control {
    display: block;
    padding: var(--mantine-spacing-xs) var(--mantine-spacing-md);
    border-radius: var(--mantine-radius-md);
    width: 15vw;
    font-weight: 500;
    text-align: center;
    position: relative;
    top: -27px;
}

.controlName {
    margin-top: 8px;
    padding: 5px 10px; 
    font-size: 2vw; 
    color: #000000; 
    transition: color 0.3s ease; 

}

.control:hover .controlName {
    color: #ffffff; 
}

.control:hover {
    background-color: var(--mantine-color-aspa-brown-4);
    max-height: 60px;
}

.activeLink {
    background: black;
    border-radius: 50%;
    width: 100px; 
    height: 100px; 
    display: flex;
    align-items: center;
    justify-content: center;
    color: white;
    font-weight: bold;
    position: absolute;
    top: 10px; 
    left: 50%;
    transform: translateX(-50%);
    position: relative;
}

.activeLink::before {
    content: '';
    position: absolute;
    width: 60px; 
    height: 30px; 
    background: white;
    border-radius: 50%;
    transform: translate(-50%, -0%) rotate(90deg); 
}

.hidden {
    display: none;
}

@media screen and (min-width: 1024px) {
    .controlName {
        font-size: 1vw; 
    }
}

@media screen and (max-width: 768px) {
    .control {
        width: 40vw; 
    }

    .controlName {
        font-size: 4vw; 
    }
}