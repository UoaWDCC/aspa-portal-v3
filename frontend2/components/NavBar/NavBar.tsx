import { useState } from 'react';
import { AppShell, Group, Burger, UnstyledButton, NavLink } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import Link from 'next/link';
import styles from './NavBar.module.css';

interface Link {
    name: string;
    href: string;
}

interface NavBarProps {
    links: Link[];
}

export function NavBar({ links }: NavBarProps) {
    const [opened, { toggle }] = useDisclosure();
    const [active, setActive] = useState(0);

    return (
        <AppShell
            navbar={{ width: 300, breakpoint: 'sm', collapsed: { desktop: true, mobile: !opened } }}
            padding="md"
        >
            <AppShell.Header className={styles.header}>
                <Group h="100%" px="md">
                    <Burger opened={opened} onClick={toggle} hiddenFrom="sm" size="sm" />
                    <Group justify="space-between" style={{ flex: 1, justifyContent: 'flex-end'}}>
                        <Group ml="xl" gap={20} visibleFrom="sm">
                            {links.map((link, index) => (
                                <Link key={index} href={link.href} passHref legacyBehavior>
                                    <a className={index === active ? '' : styles.control} onClick={() => setActive(index)}>
                                        <UnstyledButton>
                                            <div className={`${styles.controlName} ${index === active ? styles.hidden : ''}`}>
                                                {link.name}
                                            </div>
                                            
                                            {index === active && <div className={styles.activeLink}></div>}
                                        </UnstyledButton>
                                    </a>
                                </Link>
                            ))}
                        </Group>
                    </Group>
                </Group>
            </AppShell.Header>

            <AppShell.Navbar py="md" px={4}>
                {links.map((link, index) => (
                    <NavLink
                        key={index}
                        label={link.name}
                        href={link.href}
                        active={index === active}
                        onClick={() => setActive(index)}
                    />
                ))}
            </AppShell.Navbar>
        </AppShell>
    );
}