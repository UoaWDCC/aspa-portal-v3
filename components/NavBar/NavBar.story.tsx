import { Meta, StoryObj } from '@storybook/react';
import { NavBar } from './NavBar';

const links = [
  { name: 'Home', href: '/home' },
  { name: 'Events', href: '/events' },
  { name: 'ASPA Team', href: '/contact' },
  { name: 'Profile', href: '/contact' },
];

const meta: Meta<typeof NavBar> = {
  title: 'Shared/NavBar',
  component: NavBar,
  argTypes: {},
};

export default meta;

export const Default: StoryObj<typeof NavBar> = {
  args: {
    links,
  },
};
