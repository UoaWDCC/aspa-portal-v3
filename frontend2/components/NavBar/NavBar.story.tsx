import { Meta, StoryObj } from '@storybook/react';
import { NavBar } from './NavBar';

const links = [
    { name: 'Home', icon: '/home.svg', href: '/home' },
    { name: 'Contact', icon: '/contact_us.svg', href: '/contact' },
    { name: 'Events', icon: '/event.svg', href: '/events' },
];

const meta: Meta<typeof NavBar> = {
    title: 'Shared/NavBar',
    component: NavBar,
    argTypes: {},
};

export default meta;

export const Default: StoryObj<typeof NavBar> = {
    args: {
        links,
    },
};