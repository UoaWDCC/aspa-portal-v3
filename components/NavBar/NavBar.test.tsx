import { render, screen, fireEvent } from "@/test-utils";
import { NavBar } from "./NavBar";

const links = [
  { name: "Home", href: "/home" },
  { name: "Contact", href: "/contact" },
  { name: "Events", href: "/events" },
];

describe("NavBar Component testing", () => {
  const originalInnerWidth = global.innerWidth;

  beforeEach(() => {
    global.innerWidth = originalInnerWidth;
    global.dispatchEvent(new Event("resize"));
  });

  it("should render all navigation links", () => {
    render(<NavBar links={links} />);
    links.forEach((link) => {
      expect(screen.getAllByText(link.name)[0]).toBeInTheDocument();
    });
  });

  it("highlights the active link when clicked", () => {
    render(<NavBar links={links} />);
    const homeButton = screen.getAllByText("Home")[0];
    fireEvent.click(homeButton);
    // The FloatingIndicator should move, but we can check aria-label or data-active
    expect(homeButton.closest("button")).toHaveAttribute("data-active");
  });

  it("shows the burger menu on mobile and toggles navbar", () => {
    global.innerWidth = 500;
    global.dispatchEvent(new Event("resize"));
    render(<NavBar links={links} />);
    const burger = screen.getByRole("button", { name: ""});
    expect(burger).toBeInTheDocument();
    fireEvent.click(burger);
    // After clicking, the navbar should be visible (NavLink should be in the document)
    expect(screen.getAllByText("Home")[0]).toBeInTheDocument();
  });

  it("renders correctly on desktop and mobile", () => {
    // Desktop
    global.innerWidth = 1200;
    global.dispatchEvent(new Event("resize"));
    const { rerender } = render(<NavBar links={links} />);
    expect(screen.getAllByText("Home")[0]).toBeInTheDocument();

    // Mobile
    global.innerWidth = 375;
    global.dispatchEvent(new Event("resize"));
    rerender(<NavBar links={links} />);
    expect(screen.getAllByText("Home")[0]).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<NavBar links={links} />);
    expect(asFragment()).toMatchSnapshot();
  });
});