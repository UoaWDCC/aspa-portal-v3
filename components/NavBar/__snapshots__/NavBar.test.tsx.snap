// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component testing matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: fontFamily;--mantine-font-family-headings: fontFamily;--mantine-color-aspa-orange-0: #fff2e4;--mantine-color-aspa-orange-1: #fbe6d3;--mantine-color-aspa-orange-2: #f0ccaa;--mantine-color-aspa-orange-3: #e5af7e;--mantine-color-aspa-orange-4: #dc9758;--mantine-color-aspa-orange-5: #d78840;--mantine-color-aspa-orange-6: #d68032;--mantine-color-aspa-orange-7: #bd6d24;--mantine-color-aspa-orange-8: #a9611d;--mantine-color-aspa-orange-9: #945212;--mantine-color-aspa-brown-0: #faf5ef;--mantine-color-aspa-brown-1: #f0e8e0;--mantine-color-aspa-brown-2: #e2cdbb;--mantine-color-aspa-brown-3: #d5b192;--mantine-color-aspa-brown-4: #c9996f;--mantine-color-aspa-brown-5: #c38a59;--mantine-color-aspa-brown-6: #c1834d;--mantine-color-aspa-brown-7: #a96f3d;--mantine-color-aspa-brown-8: #986335;--mantine-color-aspa-brown-9: #845429;--mantine-color-aspa-gray-0: #f4f5f5;--mantine-color-aspa-gray-1: #e7e7e7;--mantine-color-aspa-gray-2: #cdcdcd;--mantine-color-aspa-gray-3: #b1b1b1;--mantine-color-aspa-gray-4: #999999;--mantine-color-aspa-gray-5: #8a8a8a;--mantine-color-aspa-gray-6: #838383;--mantine-color-aspa-gray-7: #707071;--mantine-color-aspa-gray-8: #636466;--mantine-color-aspa-gray-9: #52575b;}:root[data-mantine-color-scheme="dark"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-8);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-9);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.15);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.2);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-3);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-outline-hover: rgba(220, 151, 88, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-8);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-9);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.15);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.2);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-3);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-outline-hover: rgba(201, 153, 111, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-8);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-9);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.15);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.2);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-3);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-outline-hover: rgba(153, 153, 153, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-filled);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-7);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.1);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.12);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline-hover: rgba(214, 128, 50, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-filled);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-7);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.1);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.12);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline-hover: rgba(193, 131, 77, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-filled);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-7);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.1);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.12);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline-hover: rgba(131, 131, 131, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="inline"
  >
    :root{--app-shell-navbar-width:calc(18.75rem * var(--mantine-scale));--app-shell-navbar-offset:calc(18.75rem * var(--mantine-scale));--app-shell-padding:var(--mantine-spacing-md);}@media(min-width: 48em){:root{--app-shell-navbar-transform:translateX(calc(var(--app-shell-navbar-width) * -1));--app-shell-navbar-transform-rtl:translateX(var(--app-shell-navbar-width));--app-shell-navbar-offset:0px !important;}}@media(max-width: 47.99375em){:root{--app-shell-navbar-width:100%;--app-shell-navbar-offset:0px;--app-shell-navbar-transform:translateX(calc(var(--app-shell-navbar-width) * -1));--app-shell-navbar-transform-rtl:translateX(var(--app-shell-navbar-width));}}
  </style>
  <div
    class="m_89ab340 mantine-AppShell-root"
    data-resizing="true"
    style="--app-shell-transition-duration: 200ms; --app-shell-transition-timing-function: ease;"
  >
    <header
      class="m_3b16f56b mantine-AppShell-header right-scroll-bar-position header"
      data-with-border="true"
      style="--app-shell-header-z-index: 100;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap; width: 100%; padding-inline: var(--mantine-spacing-md); height: 100%;"
      >
        <a
          href="/"
        >
          <img
            alt="ASPA Logo"
            class="logoCircle"
            data-nimg="1"
            decoding="async"
            height="37"
            loading="lazy"
            src="/_next/image?url=%2Faspa_logo.png&w=96&q=75"
            srcset="/_next/image?url=%2Faspa_logo.png&w=48&q=75 1x, /_next/image?url=%2Faspa_logo.png&w=96&q=75 2x"
            style="color: transparent;"
            width="37"
          />
        </a>
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <button
            aria-label="Toggle menu"
            class="mantine-focus-auto m_fea6bf1a mantine-Burger-root m_87cf2631 mantine-UnstyledButton-root mantine-hidden-from-sm"
            data-size="sm"
            style="--burger-size: var(--burger-size-sm);"
            type="button"
          >
            <div
              class="m_d4fb9cad mantine-Burger-burger"
              data-reduce-motion="true"
            />
          </button>
          <div
            class="m_4081bf90 mantine-Group-root mantine-visible-from-sm"
            style="--group-gap: calc(1.25rem * var(--mantine-scale)); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
          >
            <div
              class="root"
            >
              <a
                href="/home"
              >
                <button
                  class="mantine-focus-auto control m_87cf2631 mantine-UnstyledButton-root"
                  data-active="true"
                  type="button"
                >
                  <div
                    class="controlName hidden"
                  >
                    Home
                  </div>
                </button>
              </a>
              <a
                href="/contact"
              >
                <button
                  class="mantine-focus-auto control m_87cf2631 mantine-UnstyledButton-root"
                  type="button"
                >
                  <div
                    class="controlName "
                  >
                    Contact
                  </div>
                </button>
              </a>
              <a
                href="/events"
              >
                <button
                  class="mantine-focus-auto control m_87cf2631 mantine-UnstyledButton-root"
                  type="button"
                >
                  <div
                    class="controlName "
                  >
                    Events
                  </div>
                </button>
              </a>
              <div
                class="m_96b553a6 mantine-FloatingIndicator-root"
                style="--transition-duration: 400ms; transform: translateY(-2px) translateX(-2px); width: 0px; height: 0px;"
              >
                <div
                  class="indicator"
                  style="--rotation: 0deg;"
                >
                  8
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <nav
      class="m_45252eee mantine-AppShell-navbar"
      data-with-border="true"
      style="--app-shell-navbar-z-index: calc(100 + 1); padding-inline: calc(0.25rem * var(--mantine-scale)); padding-block: var(--mantine-spacing-md);"
    >
      <a
        class="mantine-focus-auto m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
        data-active="true"
        href="/home"
      >
        <div
          class="m_f07af9d2 mantine-NavLink-body"
        >
          <span
            class="m_1f6ac4c4 mantine-NavLink-label"
          >
            Home
          </span>
          <span
            class="m_57492dcc mantine-NavLink-description"
            data-active="true"
          />
        </div>
      </a>
      <a
        class="mantine-focus-auto m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
        href="/contact"
      >
        <div
          class="m_f07af9d2 mantine-NavLink-body"
        >
          <span
            class="m_1f6ac4c4 mantine-NavLink-label"
          >
            Contact
          </span>
          <span
            class="m_57492dcc mantine-NavLink-description"
          />
        </div>
      </a>
      <a
        class="mantine-focus-auto m_f0824112 mantine-NavLink-root m_87cf2631 mantine-UnstyledButton-root"
        href="/events"
      >
        <div
          class="m_f07af9d2 mantine-NavLink-body"
        >
          <span
            class="m_1f6ac4c4 mantine-NavLink-label"
          >
            Events
          </span>
          <span
            class="m_57492dcc mantine-NavLink-description"
          />
        </div>
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
